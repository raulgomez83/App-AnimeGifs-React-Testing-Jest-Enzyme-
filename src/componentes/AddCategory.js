import React , {useState} from 'react'
import PropTypes from "prop-types"


export const AddCategory = ({setCategories}) => {
   const [inputValue, setInputValue]= useState("")

   const handleInputChange=(e)=>{
       setInputValue(e.target.value)
   }

   const handleSubmit =(e)=>{
       e.preventDefault( )
     if(inputValue.trim().length>2){
       setCategories(newCategories=>[inputValue,...newCategories,])
   }}

    return (
        <div>
        <form onSubmit={handleSubmit}>
          <input type="text" value={inputValue} onChange={handleInputChange}></input>
          </form> 
          </div>
    )
}

AddCategory.propTypes={
    setCategories:PropTypes.func.isRequired
}