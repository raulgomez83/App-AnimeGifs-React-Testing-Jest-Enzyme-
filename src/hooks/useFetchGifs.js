import { useState,useEffect } from "react"
import {getGifs} from "../helpers"  

export  const useFetchGifs = (category) => {
 const [state, setState] = useState({
    data:[],
    loading:true
  });
          //useEffect hace que la funciÃ³n se ejecute cuando el componente es renderizado por primera vez
    useEffect( ()=>{ 
        getGifs(category)
        .then(images=>{
           setState({
               data:images,
               loading:false
            });
     })
    },[category]) ;
return state;  //state  {data:[], loading:true}
}

