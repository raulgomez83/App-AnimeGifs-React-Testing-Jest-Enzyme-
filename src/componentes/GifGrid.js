import React from 'react'
import PropTypes from 'prop-types'

import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'


export const GifGrid = ({category}) => {

    const {data:images,loading} = useFetchGifs(category);
      return (
      <>
       <h3>{category}</h3>
       {loading }
       <div className="card-grid" >
       {
       images.map(img =>(<GifGridItem key={img.id} {...img}/>))
        }
       </div>
        </>
    )
}
 GifGrid.propTypes={
     category:PropTypes.string.isRequired
 }