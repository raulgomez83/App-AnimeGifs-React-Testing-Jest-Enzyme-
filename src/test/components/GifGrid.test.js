import React from "react"
import { shallow} from "enzyme";
import { GifGrid } from "../../componentes/GifGrid"
import "@testing-library/jest-dom"
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs")


describe('Testing GifGrid component', () => {

    const category="Naruto";
    test('should component works', () => {

        useFetchGifs.mockReturnValue({
            data:[],
            loading:true,
        })
        const wrapper=shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot();
    })
    test('should show items to charge image useFetchGifs', () => {
        const gifs=[{
            id:"123",
            url:"https://localhost/hueyd.jpg",
            title:"aonechuzri",
        }];
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false,
        })
        const wrapper=shallow(<GifGrid category={category} />)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("p").exists()).toBe(false);
        console.log(wrapper.find("GifGridItem"));
        console.log(gifs);
        expect(wrapper.find("GifGridItem").length).toBe(gifs.length);//deber√≠a ser uno en vez
    })

})
