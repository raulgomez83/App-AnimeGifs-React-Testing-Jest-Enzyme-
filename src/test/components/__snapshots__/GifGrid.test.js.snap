// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGrid component should component works 1`] = `
<Fragment>
  <h3>
    Naruto
  </h3>
  <Component />
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Testing GifGrid component should show items to charge image useFetchGifs 1`] = `
<Fragment>
  <h3>
    Naruto
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="123"
      key="123"
      title="aonechuzri"
      url="https://localhost/hueyd.jpg"
    />
  </div>
</Fragment>
`;
